### PR TITLE
Bump go mod version

### PR DESCRIPTION
**Pull request by ndbaker1**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Addresses govulncheck vulnerabilities with go version and project dependencies

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module k8s.io/cloud-provider-aws
 
-go 1.22.5
+go 1.22.7
```
Comment (kmala) [inline go.mod:3]: can you also update the other places like dockerfile, cloud build etc where go version is used.

Comment (ndbaker1) [reply go.mod:3]: ack done, not seeing any other places where the version pops up

Comment (kmala) [reply go.mod:3]: can you also update https://github.com/kubernetes/cloud-provider-aws/blob/release-1.30/.ko.yaml and we should be good

Comment (ndbaker1) [reply go.mod:3]: ah hidden dotfiles 😠
updated, i'll do the same to PRs for other branches after this one's merged

Comment (kmala) [reply go.mod:3]: can you cherry pick https://github.com/kubernetes/cloud-provider-aws/pull/1019/files to fix the build issues

**Pull request merged**